File: api-dot-net/Services/CodeExecution/Dto/CodeExecutionResult.cs
using CjsApi.Dto.ResponseDto;

namespace CjsApi.Services.CodeExecution.Dto
{

    public class CodeExecutionResult
    {
        public string Output { get; set; } = string.Empty;
        public string Error { get; set; } = string.Empty;
        public int ExitCode { get; set; }
        public bool TimedOut { get; set; }

        public List<TestCaseResultDto> TestCaseResults {get;set;} = new();
    }
}

File: java-api/cjs-service/src/main/java/com/cjs/cjs_service/dto/CodeExecutionResult.java
package com.cjs.cjs_service.dto;

import com.cjs.cjs_service.model.SubmissionStatus;
import com.fasterxml.jackson.annotation.JsonInclude;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CodeExecutionResult {

    private String output = "";
    private String error = "";
    private int exitCode;
    private boolean timedOut;

    private SubmissionStatus submissionStatus = SubmissionStatus.PENDING;

    private List<TestCaseResultDto> testCaseResults = new ArrayList<>();


}

